### ISSUE TITLE
[feat]long polling mode support distributed service

### ISSUE THREAD

**Issue by ralfbawg** [googollee/go-socket.io#463]
websocket is full state,but long polling is not,so if we use socket.io in a distributed system by inconsistent hash,the same sid polling connection serve by diff server 

connection info should store via redis or other middleware to support distributed service

**Comment by googollee** [comment]
No, it's not a good feature.

When using the cluster, it's a common pattern to scale up concurrent connections. In socket.io/engine.io, one session only has 2 connections at most, one for reading and another for writing. There is no scalable problem for one session. Clusters won't help here. To support concurrent sessions, it's easy to use consistent hashing and route all requests with the same sid to the same host. [Consistent hashing](https://en.wikipedia.org/wiki/Consistent_hashing) makes sure most of the connections won't remap when adding new nodes to the cluster.

If allowing connections of one session with different hosts, also makes communication complex too. When writing a message, a host needs to know where the writing connection is. The writing connection may not exist at that time, so the message should be stored somewhere and be grabbed by the new writing connection. It also faces the problem of different codes between different hosts, while rolling out a new version.

In summary, it makes the framework far more complex but doesn't get any benefit. There is an easy way to do that, [Consistent hashing](https://en.wikipedia.org/wiki/Consistent_hashing) routing.

**Comment by ralfbawg** [comment]
Got it,but as my words ,consistent hashing routing should also face the situation:service offline or crash down,maybe it's not our business.
